azurelinuxagent/ga/update.py

```diff
@@ -487,9 +488,44 @@ def _process_goal_state(self, exthandlers_handler, remote_access_handler):
         finally:
             self.last_incarnation = incarnation
 
+    @staticmethod
+    def __get_vmagent_upgrade_status(protocol, incarnation_changed):
+        """
+        This function gets the VMAgent upgrade status as per the last GoalState.
+        Returns: None if the last GS does not ask for requested version else VMAgentUpdateStatus
+        """
+        upgrade_status = None
+
+        try:
+            agent_manifests, _ = protocol.get_vmagent_manifests()
+            # Expectation here is that there will only be one manifest per family passed down from CRP.
+            # If there are multiple, we pick the first one
+            manifest = next(m for m in agent_manifests if m.family == conf.get_autoupdate_gafamily())
+            if manifest.is_requested_version_available:
+                if CURRENT_VERSION == manifest.requested_version:
+                    status = VMAgentUpdateStatuses.Success
+                    code = 0
+                else:
+                    status = VMAgentUpdateStatuses.Error
+                    code = -1
+                upgrade_status = VMAgentUpdateStatus(expected_version=manifest.version, status=status, code=code)
+        except StopIteration:
```

Comment:
could we have this except clause enclose only the actual iteration? if at some point another iteration is added to this code this could lead to problems

Reply:
Done